clusterloader2/cmd/clusterloader.go

```diff
@@ -307,22 +304,46 @@ func main() {
 
 	testReporter := test.CreateSimpleReporter(path.Join(clusterLoaderConfig.ReportDir, "junit.xml"), "ClusterLoaderV2")
 	testReporter.BeginTestSuite()
+
+	var testScenarios []api.TestScenario
 	if testSuiteConfigPath != "" {
 		testSuite, err := config.LoadTestSuite(testSuiteConfigPath)
 		if err != nil {
 			klog.Exitf("Error while reading test suite: %v", err)
 		}
+
 		for i := range testSuite {
-			clusterLoaderConfig.TestScenario = testSuite[i]
-			runSingleTest(f, prometheusFramework, testReporter)
+			testScenario := testSuite[i]
+			testScenarios = append(testScenarios, testScenario)
```

Comment:
why do you need this?
why not simply:

testScenarios = ([]api.TestScenario)testSuite